DFC.ServiceTaxonomy.JobProfiles.Module/Repositories/IJobProfileRepository.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Linq.Expressions;
+using YesSql;
+using YesSql.Indexes;
+
+namespace DFC.ServiceTaxonomy.JobProfiles.Module.Repositories
+{
+    public interface IJobProfileRepository<T> : IDisposable where T : MapIndex
```

Comment:
The implementation suggests it can be any Index because it's generic and can be used with any index by the consumer, so would it be worth changing it to something like IRepository<T> and Repository<T>? Could also be useful in other projects within STAX, could even create a separate class lib called DFC.ServiceTaxonomy.Common and move it there perhaps?